src/price.rs

```diff
@@ -26,4 +27,48 @@ impl Contract {
             panic!("Market for token: {} was not found", token);
         })
     }
+
+    pub fn get_liquidation_price(
+        &self,
+        sell_token_amount: U128,
+        sell_token_price: U128,
+        buy_token_price: U128,
+        leverage: U128,
+        borrow_fee: U128,
+        swap_fee: U128,
+    ) -> WBigDecimal {
+
+        let collateral_usd = BigDecimal::from(sell_token_amount) * BigDecimal::from(sell_token_price);
+        
+        let buy_amount = collateral_usd / BigDecimal::from(buy_token_price);
+
+        let borrow_amount = collateral_usd * BigDecimal::from(leverage);
+
+        (BigDecimal::from(buy_token_price) - (collateral_usd - BigDecimal::from(borrow_fee) * borrow_amount) / buy_amount + BigDecimal::from(sell_token_amount) * BigDecimal::from(swap_fee)).into()
```

Comment:
@vlad-sundunchan 

calculation pseudo-code, pls ensure that all input parameters are used in correct dimension
```
collateral_usd = sell_token_amount * sell_token_price
position_amount_usd = collateral_usd * leverage 
volatility_rate = 95%
borrow_amount = collateral_usd * (leverage - 1)
buy_amount = position_amount_usd / buy_token_price

LP = (position_amount_usd - volatility_rate * collateral_usd + borrow_amount * borrow_fee + position_amount_usd * swap_fee) / buy_amount
```

interface
```
pub fn calculate_liquidation_price(
    sell_token_amount: U128,
    sell_token_price: Price,
    buy_token_price: Price,
    leverage: U128,
    borrow_fee: U128,
    swap_fee: U128,
) -> Price // ticker_id = usd
{
...
}
```